### PR TITLE
Buster

### PR DESCRIPTION
**Pull request by yalh76**
## Problem
- *Fix upgrade for buster:*

## Solution
- *Apply example_ynh*
- *On buster `hostapd` fails to start if there is no wifi interface.
and if `hostapd` service fails, `ynh-hotspot` service fails too
and during upgrde, there was: 
https://github.com/labriqueinternet/hotspot_ynh/blob/ec919eb58fdc9c358ddbe17558874ae968a7f527/scripts/upgrade#L146*

that make the upgrade fails

So I added the same `$wifi_device` detection in the upgrade and the restore script

## PR Status
- [x] Code finished.
- [x] Tested with Package_check.
- [x] Fix or enhancement tested.
- [x] Upgrade from last version tested.
- [x] Can be reviewed and tested.

## Package_check results
---
[Build Status/badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/hotspot_ynh%20PR64%20(yalh76)/)  

Own CI Results for Revision: d5d9fcbcb9f208292b24ace18e1a8a2945b1f34a 
### Stretch
image

### Buster
image

### CODE REVIEW

**Comment by yalh76** [comment]
ping @keomabrun @kay0u @alexAubin for hotspot ready for buster

File: scripts/install [hunk 1]
```diff
@@ -296,34 +297,42 @@ systemctl disable hostapd
...
 
-# register the service
-yunohost service add $service_name --description "Creates a Wi-Fi access point" --need_lock
+#=================================================
+# INTEGRATE SERVICE IN YUNOHOST
+#=================================================
+ynh_script_progression --message="Integrating service in YunoHost..."
 
-# enable and start the service if device is present
+yunohost service add $service_name --description "Creates a Wi-Fi access point" --log_type "systemd" --need_lock
```
Comment (alexAubin) [inline scripts/install:312]: Zblerg, not sure why adding --log_type=systemd ... in fact it's kind of a deprecated option nowadays :/

```suggestion
yunohost service add $service_name --description "Creates a Wi-Fi access point" --need_lock
```

File: scripts/upgrade [hunk 2]
```diff
@@ -108,64 +132,105 @@ ynh_install_app_dependencies "$pkg_dependencies" "$packages"
...
 chown -R $app: ${final_path}/
 chmod -R 0644 ${final_path}/*
 find ${final_path}/ -type d -exec chmod +x {} \;
 
+#=================================================
+# INTEGRATE SERVICE IN YUNOHOST
+#=================================================
+ynh_script_progression --message="Integrating service in YunoHost..."
+
+yunohost service add $service_name --description "Creates a Wi-Fi access point" --log_type "systemd" --need_lock
```
Comment (alexAubin) [inline scripts/upgrade:202]:
```suggestion
yunohost service add $service_name --description "Creates a Wi-Fi access point" --need_lock
```

File: scripts/restore [hunk 3]
```diff
@@ -64,86 +67,110 @@ if [[ $firmware_nonfree -eq 1 ]]; then
...
 systemctl enable $service_name.service
 
 #=================================================
-# ADVERTISE SERVICE IN ADMIN PANEL
+# INTEGRATE SERVICE IN YUNOHOST
 #=================================================
+ynh_script_progression --message="Integrating service in YunoHost..."
 
-yunohost service add $service_name --description "Creates a Wi-Fi access point" --need_lock
+yunohost service add $service_name --description "Creates a Wi-Fi access point" --log_type "systemd" --need_lock
```
Comment (alexAubin) [inline scripts/restore:139]:
```suggestion
yunohost service add $service_name --description "Creates a Wi-Fi access point" --need_lock
```

File: scripts/install (hunk 1 above)
Comment (yalh76) [reply scripts/install:312]: adding `--log_type=systemd` allows to directly have journalctl informations in the service tab. For example for pleroma:

https://github.com/YunoHost-Apps/pleroma_ynh/blob/e8a4a37c2fa35255d0ca3927df8da09e3c5829b6/scripts/install#L256

image

Comment (alexAubin) [reply scripts/install:312]: This was indeed necessary in the past, but now yunohost will return the journalctl logs by default, no need to specify that option (or if you do need it to work, it should be considered as a bug)

Comment (yalh76) [reply scripts/install:312]: thanks, I didn't knew that. If you can merge https://github.com/labriqueinternet/hotspot_ynh/pull/65 to solve that. I don't have rights on https://github.com/labriqueinternet

Comment (agentcobra) [reply scripts/install]: @yalh76 it's merged

**Comment by yalh76** [comment]
I let you guys merge this PR when you want ;)

**Comment by alexAubin** [comment]
Yolo then, thanks again !

**Pull request merged**